src/window.rs

```diff
@@ -710,11 +725,30 @@ pub trait Window: AsAny + Send + Sync {
     ///
     /// ## Platform-specific
     ///
-    /// - **iOS:** Returns the [`PhysicalSize`] of the window in screen space coordinates.
     /// - **Web:** Returns the size of the canvas element. _Note: this returns the same value as
     ///   [`Window::surface_size`]._
     fn outer_size(&self) -> PhysicalSize<u32>;
 
+    /// The area of the surface that is unobstructed.
+    ///
+    /// On some devices, especially mobile devices, the screen is not a perfect rectangle, and may
+    /// have rounded corners, notches, bezels, and so on. When drawing your content, you usually
+    /// want to draw your background and other such unimportant content on the entire surface, while
+    /// you will want to restrict important content such as text, interactable or visual indicators
+    /// to the part of the screen that is actually visible; for this, you use the safe area.
+    ///
+    /// The safe area is a rectangle that is defined relative to the origin at the top-left corner
+    /// of the surface, and the size extending downwards to the right. The area will not extend
+    /// beyond [the bounds of the surface][Window::surface_size].
+    ///
+    /// If the entire content of the surface is visible, this returns `((0, 0), surface_size)`.
```

Comment:
I think this should note that it doesn't account for occluded areas.

I briefly thought about if we should account for occluded areas, but users need both separately.
Because safe area can define how you layout thing in your application, but occluded should not.

Reply:
Safe area is area without occlusion from hardware elements. Like notches, etc. It's nothing more that that. It's also the case for e.g. macOS decorations in some cases, but it's _winit specific_.

Regular occlusion by other applications is not it.

Virtual keyboards just resize your entire application on some platforms, so it should be dealt with on case by case basis. On Wayland, you're resized to not draw inside the virtual keyboard exclusive area. On ios/android, probably something else.

Reply:
Added comment to this effect